pkg/sql/multitenant_admin_function_test.go

```diff
@@ -63,148 +63,233 @@ func createSecondaryTenantDB(
+		// If set, the test query must return these results for secondary tenants.
+		expectedSecondaryResult [][]string
+		// If set, the test query must fail with this error message for secondary tenants.
+		expectedSecondaryErrorMessage string
+		// Used for tests that have SPLIT AT as a prereq (eg UNSPLIT AT).
+		allowSplitAndScatter bool
+		// Used as a stop-gap to bypass the system tenant check and test incomplete functionality.
 		skipSQLSystemTenantCheck bool
 	}{
 		{
-			desc:                 "ALTER RANGE x RELOCATE LEASE",
-			query:                "ALTER RANGE (SELECT min(range_id) FROM [SHOW RANGES FROM TABLE t]) RELOCATE LEASE TO 1;",
-			expectedErrorMessage: errorutil.UnsupportedWithMultiTenancyMessage,
+			desc:                          "ALTER RANGE x RELOCATE LEASE",
+			query:                         "ALTER RANGE (SELECT min(range_id) FROM [SHOW RANGES FROM TABLE t]) RELOCATE LEASE TO 1;",
+			expectedSecondaryErrorMessage: errorutil.UnsupportedWithMultiTenancyMessage,
 		},
 		{
 			desc:                     "ALTER RANGE x RELOCATE LEASE SkipSQLSystemTenantCheck",
 			query:                    "ALTER RANGE (SELECT min(range_id) FROM [SHOW RANGES FROM TABLE t]) RELOCATE LEASE TO 1;",
+			expectedSystemResult:     [][]string{{"54", "/Table/53", "ok"}},
+			expectedSecondaryResult:  [][]string{{"55", "", rangeErrorMessage}},
 			skipSQLSystemTenantCheck: true,
 		},
 		{
-			desc:                 "ALTER RANGE RELOCATE LEASE",
-			query:                "ALTER RANGE RELOCATE LEASE TO 1 FOR (SELECT min(range_id) FROM [SHOW RANGES FROM TABLE t]);",
-			expectedErrorMessage: errorutil.UnsupportedWithMultiTenancyMessage,
+			desc:                          "ALTER RANGE RELOCATE LEASE",
+			query:                         "ALTER RANGE RELOCATE LEASE TO 1 FOR (SELECT min(range_id) FROM [SHOW RANGES FROM TABLE t]);",
+			expectedSecondaryErrorMessage: errorutil.UnsupportedWithMultiTenancyMessage,
 		},
 		{
 			desc:                     "ALTER RANGE RELOCATE LEASE SkipSQLSystemTenantCheck",
 			query:                    "ALTER RANGE RELOCATE LEASE TO 1 FOR (SELECT min(range_id) FROM [SHOW RANGES FROM TABLE t]);",
+			expectedSystemResult:     [][]string{{"54", "/Table/53", "ok"}},
+			expectedSecondaryResult:  [][]string{{"55", "", rangeErrorMessage}},
 			skipSQLSystemTenantCheck: true,
 		},
 		{
-			desc:                 "ALTER RANGE x RELOCATE VOTERS",
-			query:                "ALTER RANGE (SELECT min(range_id) FROM [SHOW RANGES FROM TABLE t]) RELOCATE VOTERS FROM 1 TO 1;",
-			expectedErrorMessage: errorutil.UnsupportedWithMultiTenancyMessage,
+			desc:                          "ALTER RANGE x RELOCATE VOTERS",
+			query:                         "ALTER RANGE (SELECT min(range_id) FROM [SHOW RANGES FROM TABLE t]) RELOCATE VOTERS FROM 1 TO 1;",
+			expectedSecondaryErrorMessage: errorutil.UnsupportedWithMultiTenancyMessage,
 		},
+		// TODO(ewall): Set up test to avoid "trying to add a voter to a store that already has a VOTER_FULL" error.
```

Comment:
Some tests require additional setup that I will address when working on adding the Class to the related KV AdminRequest.